tests/h/schemas/forms/accounts/reset_password_test.py

```diff
@@ -1,86 +1,128 @@
+from datetime import datetime, timedelta
+from unittest.mock import create_autospec
+
 import colander
 import pytest
-from itsdangerous import BadData, SignatureExpired
+import pytz
+from itsdangerous import BadData, SignatureExpired, URLSafeTimedSerializer
 
 from h.schemas.forms.accounts import ResetPasswordSchema
 
-pytestmark = pytest.mark.usefixtures("pyramid_config")
-
 
-@pytest.mark.usefixtures("user_model")
-class TestResetPasswordSchema:
-    def test_it_is_invalid_with_password_too_short(self, pyramid_csrf_request):
-        schema = ResetPasswordSchema().bind(request=pyramid_csrf_request)
+@pytest.mark.usefixtures("pyramid_config")
+class TestResetPasswordSchemaDeserialize:
+    def test_it_is_valid_with_a_long_password(self, schema):
+        # Yeah... our minimum password length is 2 chars. See
+        # `h.schema.forms.accounts.util`
+        schema.deserialize({"user": "*any*", "password": "aa"})
 
+    @pytest.mark.parametrize("password", ("", "a"))
+    def test_it_is_invalid_with_password_too_short(self, schema, password):
         with pytest.raises(colander.Invalid) as exc:
-            schema.deserialize({"password": "a"})
+            schema.deserialize({"user": "*any*", "password": password})
+
         assert "password" in exc.value.asdict()
 
-    def test_it_is_invalid_with_invalid_user_token(self, pyramid_csrf_request):
-        pyramid_csrf_request.registry.password_reset_serializer = (
-            self.FakeInvalidSerializer()
-        )
-        schema = ResetPasswordSchema().bind(request=pyramid_csrf_request)
+    def test_it_is_invalid_with_a_null_user(self, schema):
+        with pytest.raises(colander.Invalid) as exc:
+            schema.deserialize({"user": colander.null, "password": "*any*"})
+
+        assert "user" in exc.value.asdict()
+
+    def test_it_is_invalid_with_a_missing_user(self, schema, models):
+        models.User.get_by_username.return_value = None
+
+        with pytest.raises(colander.Invalid) as exc:
+            schema.deserialize({"user": "encoded_token", "password": "*any*"})
+
+        assert "user" in exc.value.asdict()
+
+    def test_it_is_invalid_with_invalid_user_token(self, schema, serializer):
+        serializer.loads.side_effect = BadData("Invalid token")
 
         with pytest.raises(colander.Invalid) as exc:
-            schema.deserialize({"user": "abc123", "password": "secret"})
+            schema.deserialize({"user": "INVALID_TOKEN", "password": "*any*"})
```

Comment:
What I'm trying to do here is explain:

* The `user` value is actually an encoded token
* And in _this_ test it's invalid (or meant to be)
* And in _this_ test the password doesn't matter